backend/sftp/sftp.go

```diff
@@ -507,7 +512,29 @@ func NewFs(name, root string, m configmap.Mapper) (fs.Fs, error) {
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to parse private key file")
 		}
-		sshConfig.Auth = append(sshConfig.Auth, ssh.PublicKeys(signer))
+
+		// If a public key has been specified then use that
+		if pubkeyFile != "" {
+			certfile, err := ioutil.ReadFile(pubkeyFile)
+			if err != nil {
+				return nil, errors.Wrap(err, "unable to read cert file")
+			}
+
+			pk, _, _, _, err := ssh.ParseAuthorizedKey(certfile)
+			if err != nil {
+				return nil, errors.Wrap(err, "unable to parse cert file")
+			}
+
+			// And the signer for this, which includes the private key signer
+			// This is what we'll pass to the ssh client.
+			pubsigner, err := ssh.NewCertSigner(pk.(*ssh.Certificate), signer)
```

Comment:
Can this type assertion `pk.(*ssh.Certificate)`  ever fail?

Reply:
I've been playing with this and I think I've managed to get it to fail; if a _standard_ public key is passed instead of a certificate then we get a `panic: interface conversion: ssh.PublicKey is *ssh.rsaPublicKey, not *ssh.Certificate`.  I'll wrap it in a `case` test so now it reads: `2020/09/25 07:57:42 Failed to create file system for "sftpcert:/tmp/XX": Public key file is not a certificate file: /tmp/TSTKEYS/id_rsa.pub`

Reply:
I'd say more idiomatic for a type with a single case would be this, which also saves doing the type assertion more than once (the compiler may be clever enough to figure this out - I dont know!)

```go
cert, ok := pk.(*ssh.Certificate)
if !ok {
    return nil, errors.New("public key file is not a certificate file: " + pubkeyFile)
}
pubsigner, err := ssh.NewCertSigner(cert, signer)
```